lighthouse-core/computed/computed-artifact.js

```diff
@@ -25,6 +25,8 @@ function makeComputedArtifact(computableArtifact) {
    * @return {ReturnType<C['compute_']>}
    */
   const request = (artifacts, context) => {
+    /** @type {Map<string, ArbitraryEqualityMap>} */
+    // @ts-ignore - break immutability solely for this caching-controller function.
```

Comment:
is there a `Mutable<T>` type we could put `Context['computedCache']` through? It'd be nice to not lose that type safety just for the mutability part

Reply:
> is there a `Mutable<T>` type we could put `Context['computedCache']` through? It'd be nice to not lose that type safety just for the mutability part

`Map` is a little special (as are `Array` and `Set`) because typescript has a special readonly type for it that removes all the mutating functions from instances, so `Map`s are `ReadonlyMap`s, but `ReadonlyMap`s aren't `Map`s.

`Mutable<T>` could do an unsafe cast for that, but we could just as easily do that here too. There's little enough going on in this line of code that ignoring seemed fine, but maybe the cast would be better?

Reply:
not a big deal but even the unsafe cast of ReadonlyMap<T> to Map<T> would be a huge upgrade IMO so we can catch the compatibility of T :)

Reply:
> so we can catch the compatibility of T

very good point, sold :)

I'll wait for #10461 to be landable so I can use tsc 3.7 stuff